.github/CODEOWNERS

```diff
@@ -459,6 +459,10 @@ utils/bazel/llvm-project-overlay/libc/** @llvm/pr-subscribers-libc
 /clang/lib/CodeGen/Targets/AArch64.cpp @llvm/pr-subscribers-AArch64
 /clang/include/clang/Basic/BuiltinsAArch64* @llvm/pr-subscribers-AArch64
 
+# AMDGPU
+**/*AMDGPU* @llvm/pr-subscribers-amdgpu
```

Comment:
`**/*AMDGPU*` is identical to `*AMDGPU*`.

Verified with
```
echo '*AMDGPU*' >> .gitignore
git check-ignore -v --no-index mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp
```

I think the current `pr-subscribers-$backend` teams are for files absolutely related to the backend, not other files: https://discourse.llvm.org/t/input-needed-teams-for-pull-request-subscriptions/73116/55

> llvm/**/*X86* seems like it’ll grab too much, if I’m following the pattern correctly. For example, it would match `llvm/test/tools/llvm-objdump/X86, and many of the tests there have little to do with x86, other than that they need a valid disassembly target to demonstrate generic features (e.g. the --start-address flag) in llvm-objdump.
>
> More generally, those sort of patterns may match similar places for other targets in the test directories.

Reply:
Test directories should probably be covered. I'm not too worried about picking up extra tests